index.js

```diff
@@ -1,254 +1,263 @@
-            if (err) return reject(err, {})
-            resolve(components)
-          })
+        const p = new Promise(function (resolve, reject) {
+            async.seq(sortComponents, ensureComponents, (components, cb) => {
+                debug('System %s started', params.name)
+                running = components
+                cb(null, components)
+            })(function (err, components) {
+                if (err) return reject(err, {})
+                resolve(components)
+            })
         })
         return cb ? p.then(immediateCallback(cb)).catch(immediateError(cb, {})) : p
     }
 
-    function ensureComponents(components, cb) {
+    const ensureComponents = (components, cb) => {
         if (running) return cb(null, running)
         async.reduce(components.reverse(), {}, toSystem, cb)
     }
 
-    function toSystem(system, name, cb) {
+    const toSystem = (system, name, cb) => {
         debug('Inspecting compontent %s', name)
-        getDependencies(name, system, function(err, dependencies) {
+        getDependencies(name, system, (err, dependencies) => {
             if (err) return cb(err)
             startComponent(dependencies, name, system, cb)
         })
     }
 
-    function startComponent(dependencies, name, system, cb) {
+    const startComponent = (dependencies, name, system, cb) => {
         debug('Starting component %s', name)
         started.push(name)
-        var component = definitions[name].component
-        var onStarted = function(err, started) {
+        const component = definitions[name].component
+        const onStarted = (err, started) => {
             if (err) return cb(err)
             setProp(system, name, started)
             debug('Component %s started', name)
-            setImmediate(function() {
+            setImmediate(() => {
                 cb(null, system)
             })
         }
         const p = component.start(dependencies, onStarted)
         if (p && p.then) {
-          p.then(immediateCallback(onStarted)).catch(immediateError(cb))
+            p.then(immediateCallback(onStarted)).catch(immediateError(cb))
         }
     }
 
-    function stop(cb) {
+    const stop = (cb) => {
         debug('Stopping system %s', params.name)
-        var p = new Promise(function(resolve, reject) {
-          async.seq(sortComponents, removeUnstarted, stopComponents, function(cb) {
-              debug('System %s stopped', params.name)
-              running = false
-              cb()
-          })(function(err) {
-            if (err) return reject(err)
-            resolve();
-          })
+        const p = new Promise(function (resolve, reject) {
+            async.seq(sortComponents, removeUnstarted, stopComponents, function (cb) {
+                debug('System %s stopped', params.name)
+                running = false
+                cb()
+            })(function (err) {
+                if (err) return reject(err)
+                resolve();
+            })
```

Comment:
We should probably reformat with 2 spaces rather than 4. It's way more common now.